File: androidx/src/androidTest/java/me/panpf/androidx/test/graphics/BitmapTest.java
package me.panpf.androidx.test.graphics;

import android.graphics.Bitmap;
import android.graphics.Color;
import android.graphics.drawable.Drawable;
import android.support.test.runner.AndroidJUnit4;

import org.junit.Test;
import org.junit.runner.RunWith;

import me.panpf.androidx.graphics.Bitmapx;
import me.panpf.androidx.graphics.drawable.Drawablex;

@RunWith(AndroidJUnit4.class)
public class BitmapTest {

    @Test
    public void testModifyColorBitmap() {
        Bitmap sourceBitmap = Bitmapx.createByColor(100, 100, Color.parseColor("#FF0000"));

        Drawable drawable = Bitmapx.toDrawableByColor(sourceBitmap, Color.parseColor("#0000FF"));
        Bitmap bitmap = Drawablex.toBitmapWithIntrinsicSize(drawable);

        sourceBitmap.recycle();
        bitmap.recycle();
    }
}

File: androidx-kt/src/androidTest/java/me/panpf/androidxkt/test/graphics/BitmapTest.kt
package me.panpf.androidxkt.test.graphics

import android.graphics.Color
import android.support.test.runner.AndroidJUnit4
import me.panpf.androidxkt.graphics.createBitmapByColor
import me.panpf.androidxkt.graphics.drawable.toBitmapWithIntrinsicSize
import me.panpf.androidxkt.graphics.toDrawableByColor
import org.junit.Test
import org.junit.runner.RunWith

@RunWith(AndroidJUnit4::class)
class BitmapTest {

    @Test
    fun testModifyColorBitmap() {
        val sourceBitmap = createBitmapByColor(100, 100, Color.parseColor("#FF0000"))

        val drawable = sourceBitmap.toDrawableByColor(Color.parseColor("#0000FF"))
        val bitmap = drawable.toBitmapWithIntrinsicSize()

        sourceBitmap.recycle()
        bitmap.recycle()
    }
}
